dist/tools/cppcheck/check.sh

```diff
@@ -80,7 +80,7 @@ if github_annotate_is_on; then
         DETAILS="${DETAILS}     * (reason: <your reason why you think this is "
         DETAILS="${DETAILS}a false positive>) */\n"
 
-        if echo "${SEVERITY}" | grep -q "\<error\>"; then
+        if [ ${RESULT} -ne 0 ] || echo "${SEVERITY}" | grep -q "\<error\>"; then
```

Comment:
Mind adding a comment above this line to explain to the next reader why we always treat everything as error on non-zero exit code?

Reply:
Done and squashed.